dex/src/main/java/io/crate/data/join/JoinBatchIterator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.data.join;
+
+import io.crate.data.BatchIterator;
+import io.crate.data.Row;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Basic class for BatchIterator implementations used for Joins
```

Comment:
This comment doesn't really contain any content that isn't already contained in `abstract class JoinBatchIterator`

Much more interesting would be what base functionality is provided by this abstract class, and what classes that extend this class are supposed to do. How do they interact?